internal/alb/ls/rules.go

```diff
@@ -529,3 +537,53 @@ func createsRedirectLoop(listener *elbv2.Listener, r elbv2.Rule) bool {
 	}
 	return false
 }
+
+// isUnconditionalRedirect checks whether specified rule always redirects
+// We consider the rule is a unconditional redirect if
+// 1) Path is /*
+// 2) All other rule conditions are nil (ignoring the Host condition).
+// 3) RedirectConfig is not nil.
+func isUnconditionalRedirect(listener *elbv2.Listener, r elbv2.Rule) bool {
+    for _, action := range r.Actions {
+        rc := action.RedirectConfig
+        if rc == nil {
+            continue
+        }
+
+        var paths []string
+        for _, c := range r.Conditions {
+            switch aws.StringValue(c.Field) {
+            case conditions.FieldPathPattern:
+                paths = append(paths, aws.StringValueSlice(c.PathPatternConfig.Values)...)
+            case conditions.FieldHTTPRequestMethod:
+                return false
+            case conditions.FieldSourceIP:
+                return false
+            case conditions.FieldHTTPHeader:
+                return false
+            case conditions.FieldQueryString:
+                return false
```

Comment:
In go you could use a comma separated list for this:
```go
case conditions.FieldHTTPRequestMethod, conditions.FieldSourceIP, conditions.FieldHTTPHeader, conditions.FieldQueryString:
    return false
```

Reply:
Thank you @mogren - I pushed a update with your suggestion included.